app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt

```diff
@@ -151,7 +151,11 @@ constructor(private val db: FirebaseFirestore, private val auth: FirebaseAuth) :
 
   private fun deserializeUserPublic(data: Map<String, Any>): UserPublic {
     val gson = Gson()
-    return gson.fromJson(gson.toJson(data), UserPublic::class.java)
+    val json = gson.toJson(data)
+    val userPublic = gson.fromJson(json, UserPublic::class.java)
+    // Ensure default values for missing fields
+    val userReputationScore = (data["userReputationScore"] as? Number)?.toDouble() ?: 0.0
+    return userPublic.copy(userReputationScore = userReputationScore)
```

Comment:
The reason that your CI does not pass is for this : 
```
override fun getUserById(
      userId: String, onSuccess: (User) -> Unit, onFailure: (Exception) -> Unit
  ) {
    db.collection(collectionPath).document(userId).get()
        .addOnSuccessListener { document ->
          val userPublic: UserPublic =
              try {
                Log.d("UserRepositoryFirestore", " Correctly fetched UserPublic")
                deserializeUserPublic(document.data!!)
              } catch (e: Exception) {
                Log.d("UserRepositoryFirestore", "Error fetching UserPublic")
                onFailure(e)                                                          // onFailure
                return@addOnSuccessListener
              }
          db.collection(collectionPath)
              .document(userId).collection("private").document("details").get()
              .addOnSuccessListener { detailsDocument ->
                val userDetails = deserializeUserDetails(detailsDocument.data!!)
                onSuccess(User(userPublic, userDetails))                              // onSuccess
              }
              .addOnFailureListener { onSuccess(User(userPublic, null)) }             // onSuccess
        }
        .addOnFailureListener(onFailure)                                              // onFailure
  }
```
Somehow, your modification in `deserializeUserPublic` does not have the same behavior as before. Maybe try another way to deal with `userReputationScore` that could avoid an error down the line. It is calling `onFailure` instead of `onSuccess`, even though you seems to obtain the good return value.

Reply:
Yes, I can confirm this from looking at the call stack of the failing test. I've been looking into why, still at it...

Reply:
Well it seems from your new push that the old serializer works for this. I have no objections to reverting as we'll remove the default values for null fields from the deserializers when we clean the database next week, so good work finding a fix!